File: ejercicios-de-programacion/python/02-dosSumas.py
def twoSum(nums, target):
    for idx, val in enumerate(nums):
        if target - val in nums[idx + 1:]:
            return [idx, nums[idx + 1:].index(target - val) + (idx + 1)]

#            0  1  2   3
demoNums = [3,2,4]
demoTarget =  6

print(
    twoSum(demoNums, demoTarget)
)

File: ejercicios-de-programacion/javascript/02-dosSumas.js
/*
DOS SUMAS

Dada una matriz de enteros nums y 
un entero target, regreso índices 
de los dos números de modo que se 
sumen a target.
Puede suponer que cada entrada tendría
exactamente una solución, y no puedes 
usar el mismo elemento dos veces.

Puede devolver la respuesta en cualquier orden.

Entrada: 
números = [ 2,7,11,15 ], objetivo = 9

Salida: [ 0,1 ]

Explicación: 
Debido a los 
números [ 0 ] + números [ 1 ] = = 9, 
devolvemos [ 0, 1 ].

*/



const twoSum = function(nums, target) {
    let vals = {};

    for (let i = 0; i < nums.length; i++) {
        if (target - nums[i] in vals) {
            return [vals[target-nums[i]], i];
        } else{
            vals[nums[i]]=i;
        }
        
    }
    return[];
};

let numeros = [ 2,7,11,15 ]
let objetivo = 9

console.log(twoSum(numeros, objetivo));
